src/lazy/components/client.tsx

```diff
@@ -76,6 +72,15 @@ export function createComponentClient<C extends ComponentType<any>>({
       }, [setFallback]);
     }
 
+    if (defer === PHASE.AFTER_PAINT) {
+      // start preloading as will be needed soon
+      useEffect(() => {
```

Comment:
should we add the condition inside the hook?
I thought one of the React Hook rules was https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

Reply:
In this case `defer` is a "static" value, as cannot be changed after generating the lazy import (same as `ssr` above) so it is safe to render the hook conditionally.